docs/source/scheduling.rst

```diff
@@ -219,3 +231,60 @@ Other libraries like ipyparallel_ and mpi4py_ also supply
 
 .. _ipyparallel: https://ipyparallel.readthedocs.io/en/latest/examples/Futures.html#Executors
 .. _mpi4py: https://mpi4py.readthedocs.io/en/latest/mpi4py.futures.html
+
+
+Standalone Python scripts
+-------------------------
+
+Some care needs to be taken when running Dask schedulers in a standlone Python script.
+Specifically, when using the single-machine multiprocessing scheduler or the local distributed
+scheduler, Dask will create additional Python processes. As part of Python's normal subprocess
+initialization, Python will import the contents of the script in every child process that is created
+(this is true for any Python code where child processes are created -- not just in Dask).
+This import initialization can lead to subprocesses recursively creating other subprocesses
+and eventually an error is raised.
+
+.. dropdown:: Common error encountered
+
+   .. code-block:: python
+
+      An attempt has been made to start a new process before the
+      current process has finished its bootstrapping phase.
+
+      This probably means that you are not using fork to start your
+      child processes and you have forgotten to use the proper idiom
+      in the main module:
+
+         if __name__ == '__main__':
+               freeze_support()
+               ...
+
+      The "freeze_support()" line can be omitted if the program
+      is not going to be frozen to produce an executable.
+
+To avoid this types of error, you should place your Dask code inside a ``if __name__ == "__main__":``
```

Comment:
My only suggestion is that, to me, "your Dask code" is a little vague. There is lots of dask-y code that probably shouldn't be in the `if __name__ == "__main__"` block, including auxiliary functions, things to map_partitions, etc. I'm not sure the best way to make the distinction to users though.

Reply:
How about

```diff
diff --git a/docs/source/scheduling.rst b/docs/source/scheduling.rst
index 85c5c06cf..60833f5a2 100644
--- a/docs/source/scheduling.rst
+++ b/docs/source/scheduling.rst
@@ -262,8 +262,10 @@ and eventually an error is raised.
       The "freeze_support()" line can be omitted if the program
       is not going to be frozen to produce an executable.

-To avoid this types of error, you should place your Dask code inside a ``if __name__ == "__main__":``
-block to ensure subprocesses are only created when your script is run as the main program.
+To avoid this types of error, you should place any Dask code that create subprocesses
+(for example, all ``compute()`` calls that use the multiprocessing scheduler, or when creating
+a local distributed cluster) inside a ``if __name__ == "__main__":`` block. This ensures
+subprocesses are only created when your script is run as the main program.

 For example, running ``python myscript.py`` with the script below will raise an error:

```

Reply:
Seems great to me!